src/map/mob.cpp

```diff
@@ -1961,24 +1968,36 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 	}
 
 	// Normal attack / berserk skill is only used when target is in range
-	if (battle_check_range(&md->bl, tbl, md->status.rhw.range) && !(md->sc.option&OPTION_HIDE))
-	{	//Target within range and able to use normal attack, engage
-		if (md->ud.target != tbl->id || md->ud.attacktimer == INVALID_TIMER) 
-		{ //Only attack if no more attack delay left
-			if(tbl->type == BL_PC)
-				mob_log_damage(md, tbl, 0); //Log interaction (counts as 'attacker' for the exp bonus)
-
-			if( !(mode&MD_RANDOMTARGET) )
-				unit_attack(&md->bl,tbl->id,1);
-			else { // Attack once and find a new random target
-				int search_size = (view_range < md->status.rhw.range) ? view_range : md->status.rhw.range;
-				unit_attack(&md->bl, tbl->id, 0);
-				if ((tbl = battle_getenemy(&md->bl, DEFAULT_ENEMY_TYPE(md), search_size))) {
-					md->target_id = tbl->id;
-					md->min_chase = md->db->range3;
+	if (battle_check_range(&md->bl, tbl, md->status.rhw.range))
+	{
+		if (!(md->sc.option&OPTION_HIDE)) {
+			// Target within range and able to use normal attack, engage
```

Comment:
```suggestion
			// Only attack if no more attack delay left
```

Reply:
The purpose of the comment is to explain the situation under which you get in here, not to comment on the next check. It needs to be clear that this branch is for situations where a monster can do normal attacks. This is so that if any new state is added that prevents monsters from doing normal attacks, it is easy to see it would need to be added into this branch. We shouldn't remove it.

Reply:
But you do not know, if you are able to use a normal attack (yet). That is only decided after a few checks below...
So the comment gives of a wrong impression.

Reply:
Nono, the check if the monster can do normal attacks is this:
`if (!(md->sc.option&OPTION_HIDE)) {`

Right now it only checks for Hide as we only know this situation (the check was already there I just moved it out of the main check so we can have a proper "else" added here.